### PR TITLE
Add backwards compatible checkpoint loading for EMA

### PR DESCRIPTION
**Pull request by coryMosaicML**
This PR adds backwards compatibility for loading checkpoints saved with the EMA algorithm from composer 0.12.1 and earlier. 

Here is a plot showing a checkpoint saved on composer 0.12.1 resuming training on this PR branch.
Screen Shot 2023-02-27 at 10 56 55 PM

### CODE REVIEW

**Comment by mvpatel2000** [comment]
Can you please add screenshots showing you tested resuming an old checkpoint from v12 on 1 GPU?

File: composer/algorithms/ema/ema.py [hunk 1]
```diff
@@ -286,7 +286,28 @@ def state_dict(self) -> Dict[str, Any]:
                 state_dict[attribute_name] = getattr(self, attribute_name)
         return state_dict
 
-    def load_state_dict(self, state: Dict[str, Any], strict: bool = False):
+    def load_old_state_dict_format(self, state: Dict[str, Any]):
```
[the remark this answers is not in the document]
Comment (coryMosaicML) [reply composer/algorithms/ema/ema.py]: > Can you add a comment about the Composer version number that differentiates old and new?

Done.

> Is there any way to have this method just modify the state_dict to something that can be loaded with load_new_state_dict?

Yes, switched it to something similar now.

**Comment by coryMosaicML** [comment]
> Can you please add screenshots showing you tested resuming an old checkpoint from v12 on 1 GPU?

Added one at the tip.

**Review by mvpatel2000 — APPROVED** [review]
LGTM! Thanks for the hotfix

**Pull request merged**